tests/model_registry/conftest.py

```diff
@@ -467,3 +521,646 @@ def is_model_registry_oauth(request: FixtureRequest) -> bool:
 def api_server_url(admin_client: DynamicClient) -> str:
     infrastructure = Infrastructure(client=admin_client, name="cluster", ensure_exists=True)
     return infrastructure.instance.status.apiServerURL
+
+
+# =============================================================================
+# FACTORY FIXTURES
+# =============================================================================
+
+
+@pytest.fixture(scope="class")
+def resource_creation_factory(
+    admin_client: DynamicClient,
+    model_registry_cleanup_registry: ModelRegistryCleanupRegistry,
+) -> Callable[[Any, Any], Any]:
+    """A generic factory for creating and cleaning up Kubernetes resources."""
+
+    def _create_resource(resource_class: Any, config: Any, **kwargs: Any) -> Any:
+        """Helper to instantiate, create, and register cleanup for a resource."""
+        resource_name = config.name
+        LOGGER.info(f"Creating {resource_class.__name__}: {resource_name}")
+
+        try:
+            resource = resource_class(client=admin_client, name=resource_name, **kwargs)
+            resource.create()
+
+            if config.teardown:
+
+                def cleanup() -> None:
+                    try:
+                        LOGGER.info(f"Cleaning up {resource_class.__name__}: {resource.name}")
+                        resource.delete(wait=True)
+                    except Exception as e:
+                        LOGGER.warning(f"Failed to cleanup {resource_class.__name__} {resource.name}: {e}")
+
+                model_registry_cleanup_registry.register_cleanup(cleanup_func=cleanup)
+
+            return resource
+        except Exception as e:
+            LOGGER.error(f"Failed to create {resource_class.__name__} {resource_name}: {e}")
+            raise
+
+    return _create_resource
+
+
+@pytest.fixture(scope="class")
+def secret_factory(resource_creation_factory: Callable[..., Any]) -> Callable[[SecretConfig], Secret]:
+    """Factory for creating Secret resources."""
+
+    def _create_secret(config: SecretConfig) -> Secret:
```

Comment:
We don't need a internal function here. instead you can directly return resource_creation_factory()